app/enquiries/common/datahub_utils.py

```diff
@@ -187,31 +221,310 @@ def dh_company_search(company_name):
     return companies, None
 
 
-def dh_contact_search(contact_name):
+def dh_contact_search(request, access_token, contact_name, company_id):
     """
     Peforms a Contact name search using Data hub API.
 
     Returns list of subset of fields for each contact found
     """
     contacts = []
     url = settings.DATA_HUB_CONTACT_SEARCH_URL
-    payload = {"name": contact_name}
+    payload = {"name": contact_name, "company": [company_id]}
 
-    response = dh_request("POST", url, payload)
+    response = dh_request(request, access_token, "POST", url, payload)
 
     if not response.ok:
         return contacts, response.json()
 
-    for contact in response.json()["results"]:
-        contacts.append(
+    contacts = [
+        {
+            "datahub_id": contact["id"],
+            "first_name": contact["first_name"],
+            "last_name": contact["last_name"],
+            "job_title": contact["job_title"],
+            "email": contact["email"],
+            "phone": contact["telephone_number"],
+        }
+        for contact in response.json()["results"]
+    ]
+
+    return contacts, None
+
+
+def dh_contact_create(request, access_token, enquirer, company_id, primary=False):
+    """
+    Create a contact and associate with the given Company Id.
+
+    Returns created contact and error if any
+    """
+    url = settings.DATA_HUB_CONTACT_CREATE_URL
+    enquirer = enquirer.enquirer
+    payload = {
+        "first_name": enquirer.first_name,
+        "last_name": enquirer.last_name,
+        "job_title": enquirer.job_title,
+        "company": company_id,
+        "primary": primary,
+        "telephone_countrycode": enquirer.phone_country_code,
+        "telephone_number": enquirer.phone,
+        "email": enquirer.email,
+        "address_same_as_company": True,
+    }
+
+    response = dh_request(request, access_token, "POST", url, payload)
+    if not response.ok:
+        return None, response.json()
+
+    return response.json(), None
+
+
+def dh_adviser_search(request, access_token, adviser_name):
+    """
+    Peforms an Adviser search using Data hub API.
+
+    Returns list of subset of fields for each Adviser found
+    """
+    advisers = []
+    url = f"{settings.DATA_HUB_ADVISER_SEARCH_URL}"
```

Comment:
Looks like this f-string could be removed:

```suggestion
    url = settings.DATA_HUB_ADVISER_SEARCH_URL
```